pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -264,903 +277,307 @@ func TestReconcile(t *testing.T) {
-		wantOwnerReferences    []metav1.OwnerReference
-		wantServiceAccountName string
-		wantPodVolume          []corev1.Volume
-		wantSteps              []corev1.Container
+		name    string
+		taskRun *v1alpha1.TaskRun
+		wantPod *corev1.Pod
 	}{{
-		name:          "success",
-		taskRun:       taskRunSuccess,
-		wantPodLabels: map[string]string{"pipeline.knative.dev/taskRun": "test-taskrun-run-success"},
-		wantOwnerReferences: []metav1.OwnerReference{{
-			APIVersion:         "pipeline.knative.dev/v1alpha1",
-			Kind:               "TaskRun",
-			Name:               "test-taskrun-run-success",
-			Controller:         &boolTrue,
-			BlockOwnerDeletion: &boolTrue,
-		}},
-		wantServiceAccountName: "",
-		wantPodVolume:          getVolumes(),
-		wantSteps: []corev1.Container{{
-			Name:       "build-step-credential-initializer-9l9zj",
-			Image:      "override-with-creds:latest",
-			Args:       []string{},
-			WorkingDir: "/workspace",
-			Env: []corev1.EnvVar{{
-				Name:  "HOME",
-				Value: "/builder/home",
-			},
-			},
-			VolumeMounts: []corev1.VolumeMount{{
-				Name:      "workspace",
-				ReadOnly:  false,
-				MountPath: "/workspace",
-			}, {
-				Name:      "home",
-				ReadOnly:  false,
-				MountPath: "/builder/home",
-			},
-			},
-		}, {
-			Name:       "build-step-place-tools",
-			Image:      "gcr.io/k8s-prow/entrypoint@sha256:7c7cd8906ce4982ffee326218e9fc75da2d4896d53cabc9833b9cc8d2d6b2b8f",
-			Command:    []string{"/bin/cp"},
-			Args:       []string{"/entrypoint", "/tools/entrypoint"},
-			WorkingDir: "/workspace",
-			Env: []corev1.EnvVar{
-				{
-					Name:  "HOME",
-					Value: "/builder/home",
-				},
-			},
-			VolumeMounts: []corev1.VolumeMount{{
-				Name:      "tools",
-				ReadOnly:  false,
-				MountPath: "/tools",
-			}, {
-				Name:      "workspace",
-				ReadOnly:  false,
-				MountPath: "/workspace",
-			}, {
-				Name:      "home",
-				ReadOnly:  false,
-				MountPath: "/builder/home",
-			},
-			},
-		}, {
-			Name:       "build-step-simple-step",
-			Image:      "foo",
-			Args:       []string{},
-			Command:    []string{"/tools/entrypoint"},
-			WorkingDir: "/workspace",
-			Env: []corev1.EnvVar{{
-				Name:      "HOME",
-				Value:     "/builder/home",
-				ValueFrom: (*corev1.EnvVarSource)(nil),
-			}, {
-				Name:      "ENTRYPOINT_OPTIONS",
-				Value:     "{\"args\":[\"/mycmd\"],\"process_log\":\"/tools/process-log.txt\",\"marker_file\":\"/tools/marker-file.txt\"}",
-				ValueFrom: (*corev1.EnvVarSource)(nil),
-			},
-			},
-			VolumeMounts: []corev1.VolumeMount{{
-				Name:      "tools",
-				ReadOnly:  false,
-				MountPath: "/tools",
-			}, {
-				Name:      "workspace",
-				ReadOnly:  false,
-				MountPath: "/workspace",
-			}, {
-				Name:      "home",
-				ReadOnly:  false,
-				MountPath: "/builder/home",
-			},
-			},
-		}},
+		name:    "success",
+		taskRun: taskRunSuccess,
+		wantPod: tb.Pod("test-taskrun-run-success-pod-123456", "foo",
+			tb.PodAnnotation("sidecar.istio.io/inject", "false"),
```

Comment:
Do we actually care about this Istio-related annotation, or is it just a holdover from when we were translating to Build and build was translating to Pod? If we aren't using it then maybe we should go ahead and delete it now.